### PR TITLE
Add support for @lifetime(borrow ...)

### PR DESCRIPTION
**Pull request by meg-gupta**

### CODE REVIEW

**Comment by meg-gupta** [comment]
Please test with following pull request:
https://github.com/swiftlang/swift/pull/77466

@swift-ci Please test

File: Sources/SwiftParser/Expressions.swift [hunk 1]
```diff
@@ -458,6 +458,11 @@ extension Parser {
         )
       )
     case (._borrow, let handle)?:
+      fallthrough
+    case (.borrow, let handle)?:
```
Comment (ahoppen) [inline Sources/SwiftParser/Expressions.swift:462]:
```suggestion
    case (._borrow, let handle)?, (.borrow, let handle)?:
```

File: Tests/SwiftParserTest/AttributeTests.swift [hunk 2]
```diff
@@ -1053,4 +1053,30 @@ final class AttributeTests: ParserTestCase {
...
+      func derive4(ne1: NE, ne2: NE) -> NE { ne1 }
+
+      @lifetime(neOut: ne)
+      func derive5(ne: NE, neOut: inout NE) -> NE { neOut = ne }
+
+      @lifetime(neOut: borrow ne)
+      func derive6(ne: borrowing NE, neOut: inout NE) -> NE { neOut = ne }
+      """
+    )
+  }
```
Comment (ahoppen) [inline Tests/SwiftParserTest/AttributeTests.swift:1081]: Could you also add a test case for `borrow blah` in a normal expression position and one that ensure we parse `borrow(blah)`, `borrow (blah)` as function calls?

**Review by ahoppen — COMMENTED** [review]
Thanks, LGTM, just two minor comments.

**Comment by meg-gupta** [comment]
Please test with following pull request:
https://github.com/swiftlang/swift/pull/77466

@swift-ci Please test

**Review by ahoppen — APPROVED** [review]

**Comment by meg-gupta** [comment]
@swift-ci Please test

**Comment by meg-gupta** [comment]
@swift-ci smoke test windows platform

**Pull request merged**